### PR TITLE
MES-5014 Mock Users & Test Data for Home Tests

### PR DESCRIPTION
**Pull request by tomcrawleyy**
InitializeVehicleChecks will be implemented when the action has been created.

## Checklist

- [x] PR title includes the JIRA ticket number
- [x] Branch is rebased against the latest develop
- [ ] Code has been tested manually
- [x] PR link added to JIRA ticket
- [ ] One review from each scrum team
- [ ] Squashed commit contains the JIRA ticket number

### CODE REVIEW

File: mock/generate-local-journal.ts [hunk 1]
```diff
@@ -890,6 +890,234 @@ const localJournal: ExaminerWorkSchedule = {
...
+          mobileTelephone: '07654 123456',
+          primaryTelephone: '01234 567890',
+          secondaryTelephone: '04321 098765',
+          dateOfBirth: '1950-02-27',
+          ethnicityCode: 'A',
+        },
+      },
+      slotDetail: {
+        duration: 57,
+        slotId: 1014,
```
Comment (TomBillingtonUK) [inline mock/generate-local-journal.ts]: needs to be unique for each slot. It's what we use as a key for the test result

File: mock/generate-local-journal.ts [hunk 2]
```diff
@@ -890,6 +890,234 @@ const localJournal: ExaminerWorkSchedule = {
       vehicleSlotTypeCode: 7,
       examinerVisiting: false,
     },
+    {
+      booking: {
+        application: {
+          applicationId: 1284472,
+          bookingSequence: 1,
+          checkDigit: 3,
```
Comment (TomBillingtonUK) [inline mock/generate-local-journal.ts]: Should be unique for each slot ,

File: src/modules/tests/tests.effects.ts [hunk 3]
```diff
@@ -205,6 +205,14 @@ export class TestsEffects {
...
         arrayOfActions.push(new InitializeVehicleChecksCatD(startTestAction.category));
       }
+      if (
+        startTestAction.category === TestCategory.F ||
+        startTestAction.category === TestCategory.G ||
+        startTestAction.category === TestCategory.H ||
+        startTestAction.category === TestCategory.K) {
+        // TODO Replace this with Home Test action
+        arrayOfActions.push(new InitializeVehicleChecksCatD(startTestAction.category));
+      }
```
Comment (TomBillingtonUK) [inline src/modules/tests/tests.effects.ts:220]: Transmission, Route Number and Independent Driving need to be defaulted in the effect - see miro for values

File: mock/generate-local-journal.ts [hunk 4]
```diff
@@ -890,6 +890,234 @@ const localJournal: ExaminerWorkSchedule = {
...
+        centreId: 54321,
+        centreName: 'Example Test Centre',
+        costCode: 'EXTC1',
+      },
+      vehicleTypeCode: 'K',
+      vehicleSlotTypeCode: 7,
+      examinerVisiting: false,
+    },
   ],
 };
```
Comment (TomBillingtonUK) [inline mock/generate-local-journal.ts:1214]: slots need to have business details in them

File: src/modules/tests/tests.effects.ts [hunk 5]
```diff
@@ -205,6 +205,14 @@ export class TestsEffects {
         startTestAction.category === TestCategory.DE) {
         arrayOfActions.push(new InitializeVehicleChecksCatD(startTestAction.category));
```
Comment (TomBillingtonUK) [inline src/modules/tests/tests.effects.ts:210]: Need to make sure we are setting business details correctly and dispatching the right actions for that.

File: src/modules/tests/tests.effects.ts [hunk 6]
```diff
@@ -206,6 +209,15 @@ export class TestsEffects {
...
       }
+      if (
+        startTestAction.category === TestCategory.F ||
+        startTestAction.category === TestCategory.G ||
+        startTestAction.category === TestCategory.H ||
+        startTestAction.category === TestCategory.K) {
+        arrayOfActions.push(new GearboxCategoryChanged('Manual'));
+        arrayOfActions.push(new RouteNumberChanged(88));
+        arrayOfActions.push(new IndependentDrivingTypeChanged('N/A'));
```
Comment (edwardwilson) [inline src/modules/tests/tests.effects.ts:219]: Spacing seems off, might cause a linting issue

**Review by edwardwilson — APPROVED** [review]
Looks good apart from my comment, I assume lintting would fail with that so might need to be fixed before merge

File: src/modules/tests/tests.effects.ts (hunk 6 above)
Comment (tomcrawleyy) [reply src/modules/tests/tests.effects.ts:219]: Doesnt seem to be an issue, it was in that format before i edited it

**Review by rossfauldsbjss — APPROVED** [review]
looks good, assuming both Tom Billington's and Ed Wilson's comments are fixed.

**Pull request merged**